include/pmacc/math/complex/Complex.tpp

```diff
@@ -180,6 +180,53 @@ struct Abs2< ::pmacc::math::Complex<T_Type> >
     }
 };
 
+    /*  Specialize log() for complex numbers. */
+    template< typename T_Type >
+    struct Log< ::pmacc::math::Complex< T_Type > >
+    {
+        using type = T_Type;
+        using result = typename ::pmacc::math::Complex< type >::type;
+
+        HDINLINE result operator( )( ::pmacc::math::Complex< T_Type > const & other )
+        {
+            return pmMath::log( pmMath::abs( other ) ) +
+                ::pmacc::math::Complex< T_Type >(
+                    type( 0. ),
+                    type( 1. )
+                ) * pmMath::arg( other );
+        }
+    };
+
+    /*  Specialize sin( ) for complex numbers. */
+    template< typename T_Type >
+    struct Sin< ::pmacc::math::Complex< T_Type > >
```

Comment:
@BeyondEspresso this is merged by now.

as you suggested, please squash a few commits and rebase against dev